modules/chapter1/pages/section1.adoc

```diff
@@ -1,3 +1,292 @@
++
+```shell
+  pip install numpy matplotlib scikit-learn
+```
++
+image::terminal-install.png[Install packages]
+
+. In **JupyterLab**, create a new notebook, and name it **purchase-amount**.
++
+image::purchase-amount-notebook.png[purchase-amount notebook]
+
+. Use the following code to train and export a model that predicts the amount of a pruchase given the number of minutes to make such purchase in a website.
++
+```python
+# Include required packages for execution
+import numpy
+import matplotlib.pyplot as plt
+from sklearn.metrics import r2_score
+import pickle
+
+# Create data
+numpy.random.seed(2)
+# Create a random set of numbers for a mock representation of minutes spent online
+x = numpy.random.normal(3,1,100)
+# Create a random set of number for a mock representation of the purchase amount
+y = numpy.random.normal(150,40,100) / x
+
+# Divide the data between tranin and test
+# where the training set is the 80% of the data
+# while the test set is the remaining 20%
+train_x = x[:80]
+train_y = y[:80]
+
+test_x = x[80:]
+test_y = y[80:]
+
+# train a model using polynomial regression
+# The number 4 here represents a hyperparameter
+mymodel = numpy.poly1d(numpy.polyfit(train_x, train_y, 4))
+# export the model to a file
+filename = 'mymodel.pkl'
+pickle.dump(mymodel, open(filename, 'wb'))
+```
+
+. Notice the creation of a new file in your environment, the `mymodel.pkl`
++
+image::mymodel-pkl.png[Model file export]
+
+[IMPORTANT]
+====
+There are different formats and libraries to export the model, in this case we are using pickle. Other common formats are:
+
+* Protobuf
+
+* MLeap
+
+* H5
+
+* ONNX
+
+* PMML
+
+* Torch
+
+The use of either of those formats depend on the target server runtime, some of them are proven to be more eficient than others for certain type of training algorithms and model sizes.
+====
+
+== Use the model in another notebook
+
+The model can be deserialized in another notebook, and used to generate a prediction:
+
+. Create a notebook called **use-purchase-amount**
++
+image::use-purchase-amount-notebook.png[use-purchase-amount notebook create]
+
+. Use the following code to instantiate the model and request for a prediction
++
+```python
+# Include required packages for execution
+import pickle
+
+# Load models
+with open('mymodel.pkl', 'rb') as f:
+    model = pickle.load(f)
+  
+# perform a prediction using the model
+print(model(4))
+```
++
+image::prediction-with-model.png[prediction with model]
+
+[TIP]
+====
+At this moment the model can be exported and imported in other projects for its use. Normally there will be an S3 bucket or a model registry to store models and versions of such models, and instead of manually exporting the model, there would be pipelines making the model available.
+====
+
+== Use the Model in a Container
```

Comment:
This requires the student to install additional dependencies on their laptop, which is not always feasible. Since the laptop is only used for building the container that is then deployed on OCP, how about this workflow instead:
- All files in this section are stored in a github repository.
- The students import this repository into OCP as a new application that includes building the container image based on the Containerfile.
- The students can test the deployed model service as you outlined.

This has the benefit of making the student familiar with a git-based app deployment workflow.